extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudDatastore.java

```diff
@@ -88,29 +86,62 @@ public void create(UUID jobId, LegacyPortabilityJob job) throws IOException {
   }
 
   /**
-   * Finds the {@link LegacyPortabilityJob} keyed by {@code jobId} in Datastore, or null if none found.
+   * Atomically updates the {@link LegacyPortabilityJob} keyed by {@code jobId} to {@code job},
+   * in Datastore using a {@link Transaction}.
+   *
+   * TODO(rtannenbaum): Consider validating authorization state was the previous one, when updating
+   * authorization state within this transaction. Previous API allowed for passing in of a previous
+   * expected state, but we shouldn't need to pass that in, given the context of the new state we
+   * should know what comes before it.
    */
   @Override
-  public LegacyPortabilityJob find(UUID jobId) {
-    Entity entity = datastore.get(getKey(jobId));
-    if (entity == null) {
-      return null;
+  public void updateJob(UUID jobId, PortabilityJob job) throws IOException {
+    Preconditions.checkNotNull(jobId);
+    Transaction transaction = datastore.newTransaction();
+    Key key = getKey(jobId);
+
+    try {
+      Entity previousEntity = transaction.get(key);
+      if (previousEntity == null) {
+        transaction.rollback();
+        throw new IOException("Could not find record for jobId " + jobId);
```

Comment:
let's do jobId: " + jobId to distinguish parameterized values or any convention to highlight for future debugging

Reply:
Here and below.

Reply:
Good point, done